packages/yarnpkg-core/sources/index.ts

```diff
@@ -8,29 +8,32 @@ import * as semverUtils from './semverUtils';
-export {Fetcher, FetchOptions, FetchResult, MinimalFetchOptions}                                         from './Fetcher';
-export {Installer, BuildDirective, BuildType, InstallStatus, FinalizeInstallStatus}                      from './Installer';
-export {LightReport}                                                                                     from './LightReport';
-export {Linker, LinkOptions, MinimalLinkOptions}                                                         from './Linker';
-export {AllDependencies, HardDependencies, Manifest, DependencyMeta, PeerDependencyMeta}                 from './Manifest';
-export {MessageName}                                                                                     from './MessageName';
-export {CommandContext, Hooks, Plugin}                                                                   from './Plugin';
-export {Project}                                                                                         from './Project';
-export {TAG_REGEXP}                                                                                      from './ProtocolResolver';
-export {ReportError, Report}                                                                             from './Report';
-export {Resolver, ResolveOptions, MinimalResolveOptions}                                                 from './Resolver';
-export {StreamReport}                                                                                    from './StreamReport';
-export {ThrowReport}                                                                                     from './ThrowReport';
-export {VirtualFetcher}                                                                                  from './VirtualFetcher';
-export {WorkspaceResolver}                                                                               from './WorkspaceResolver';
-export {Workspace}                                                                                       from './Workspace';
-export {YarnVersion}                                                                                     from './YarnVersion';
-export {IdentHash, DescriptorHash, LocatorHash}                                                          from './types';
-export {Ident, Descriptor, Locator, Package}                                                             from './types';
-export {LinkType}                                                                                        from './types';
+export {Cache}                                                                      from './Cache';
+export {DEFAULT_RC_FILENAME, DEFAULT_LOCK_FILENAME}                                 from './Configuration';
+export {Configuration, FormatType, ProjectLookup, SettingsType}                     from './Configuration';
+export type {PluginConfiguration, SettingsDefinition}                               from './Configuration';
```

Comment:
What practical effects have the `export type`?

Reply:
`export {SomeType} from './foo';` doesn't work under `isolatedModules`.
There are 2 ways to resolve this issue: `export type {SomeType} from './foo';` or
```ts
import {SomeType} from './foo'
export {SomeType};
```